Server/ConcordiaCurriculumManager/Services/CourseService.cs

```diff
@@ -116,4 +117,55 @@ public async Task<CourseCreationRequest> InitiateCourseCreation(CourseCreationIn
 
         return courseCreationRequest;
     }
+
+    public async Task<CourseModificationRequest> InitiateCourseModification(CourseModificationInitiationDTO modification, Guid userId)
```

Comment:
The logic in this service method is incorrect. It seems what you're doing it taking the old course and modifying its attributes, then saving it back into the DB, which is not what we want (because we effectively lose the data associated to the old course before this change even gets approved). Instead, what we actually want is to create a new course row in the DB (with a new GUID, it's a different row after all), just like how we do with the new course request, but with 2 differences: First the new course we create will have its CourseState attribute set to CourseChangeProposal, and second, in the course creation initiation, we set the CourseID attribute to maxCourseId + 1, whereas here we'll want the CourseID attribute to be the same as the CourseID attribute of the original course we're proposing a modification for (to get the CourseID attribute, have this passed into the request's DTO).